rllib/evaluation/env_runner_v2.py

```diff
@@ -637,8 +638,12 @@ def _process_observations(
                     policy_id: PolicyID = episode.policy_for(agent_id)
                     policy = self._worker.policy_map[policy_id]
 
-                    # Create a fake (all-0s) observation.
-                    obs_space = policy.observation_space
+                    # Create a fake (all-0s) observation (from the original space if
+                    # available
+                    obs_space = (
+                        policy.agent_connectors[ObsPreprocessorConnector]
```

Comment:
what are you indexing agent_connectors with? the class? This line does not make sense to me.

Reply:
Yes, you can do this. It makes sense.

Reply:
what is the content of `policy.agent_connectors` ?

Reply:
I modified the __get__ method of the pipeline a while back! have a look 😄

Reply:
ObsPreprocessorConnector, Viewrequirements and State Buffer?

Reply:
oh shit :) what is the expected return value of `policy.agent_connectors[ObsPreprocessorConnector]`?

Reply:
A list! Just fixed fixed this line thanks for asking!

Reply:
So we take the first observationspreprocessor and get the original space from there.

Reply:
-> This PR is a little "drafty". I'm yet to test this and maybe will write a unit test.

Reply:
add a little bit of error handling? what if ObsPreprocessorConnector is not used?